File: 2025-1/sesion-de-clase-20/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/bo/IBaseBO.java
package pe.edu.pucp.inf30.softprog.bo;

import java.util.List;
import pe.edu.pucp.inf30.softprog.model.Estado;

/**
 *
 * @author eric
 * @param <T>
 */
public interface IBaseBO<T> {
    List<T> listar();
    T obtener(int id);
    void eliminar(int id);
    void guardar(T modelo, Estado estado);
}

File: 2025-2/sesion-de-clase-23/dotnet/SoftProgNegocio/BO/IBaseBO.cs
using PUCP.SoftProg.Modelo;
using System.Collections.Generic;

namespace PUCP.SoftProg.Negocio.BO {
    public interface IBaseBO<T> {
        List<T> Listar();
        T Obtener(int id);
        void Eliminar(int id);
        void Guardar(T modelo, Estado estado);
    }
}
